packages/svelte/src/internal/client/dom/elements/attributes.js

```diff
@@ -333,3 +333,22 @@ function srcset_url_equal(element, srcset) {
 		)
 	);
 }
+
+/**
+ * @param {HTMLImageElement} element
+ * @returns {void}
+ */
+export function handle_ff_lazy_img(element) {
+	// If we're using Firefox and the image has a lazy loading attribute, we need to
+	// apply this attribute, along with the src, after it has been appended to the document
+	// otherwise the lazy behaviour will not work due to our cloneNode heuristic for templates.
+	if (/Firefox/.test(navigator.userAgent) && element.getAttribute('loading') === 'lazy') {
```

Comment:
The same bug exists in Safari, so I think we need to either do some more reliable feature detection, or always do this.

Is there a reason to do `element.getAttribute('loading')` instead of just `element.loading`?

Reply:
this is the quick-n-dirty script i used to check safari support FYI — i ran this on https://example.com

```js
function go() {
  document.querySelector('div').style.height = '200vh';

  const img1 = new Image();
  document.body.append(img1);

  img1.loading = 'lazy'; // works, because `loading` is set first
  img1.src = 'img1.jpg';

  const img2 = new Image();
  document.body.append(img2);

  img2.src = 'img2.jpg'; // fails, because `src` is set first
  img2.loading = 'lazy';
}
```